python/pylibcudf/pylibcudf/reshape.pyx

```diff
@@ -67,3 +76,41 @@ cpdef Table tile(Table source_table, size_type count):
         c_result = cpp_tile(source_table.view(), count)
 
     return Table.from_libcudf(move(c_result))
+
+
+cpdef void table_to_array(
+    Table input_table,
+    uintptr_t ptr,
+    size_type size,
+    Stream stream=None
+):
+    """
+    Copy a table into a preallocated column-major device array.
+    Parameters
+    ----------
+    input_table : Table
+        A table with fixed-width, non-nullable columns of the same type.
+    ptr : uintptr_t
+        A device pointer to the beginning of the output buffer.
+    size : size_type
+        The total number of bytes available at `ptr`.
+        Must be at least `num_rows * num_columns * sizeof(dtype)`.
+    stream : Stream | None
+        CUDA stream on which to perform the operation.
+    """
+    if size > numeric_limits[size_type].max():
```

Comment:
This shouldn't be limited by `size_type`. Shouldn't it be `size_t`?

Reply:
Yes, done in https://github.com/rapidsai/cudf/pull/18801/commits/366005d559135f06fc88a3a67f2eb995950bc19f